os_traits/const.py

```diff
@@ -67,9 +67,55 @@
 # ref: https://en.wikipedia.org/wiki/AMD-V
 HW_CPU_X86_SVM = _CPU_X86_NS + 'svm'
 
+# All hypervisor specific features are prefixed with this namespace
+_VIRT_NS = 'virt:'
+
+# All libvirt specific features are prefixed with this namespace
+_VIRT_LIBVIRT_NS = _VIRT_NS + 'libvirt:'
+
+# All Hyper-V specific features are prefixed with this namespace
+_VIRT_HYPERV_NS = _VIRT_NS + 'hyperv:'
+
+_VIRT_HYPERV_VM_GENERATION = _VIRT_HYPERV_NS + 'vm_generation:'
+VIRT_HYPERV_VM_GENERATION_1 = -_VIRT_HYPERV_VM_GENERATION + '1'
+VIRT_HYPERV_VM_GENERATION_2 = -_VIRT_HYPERV_VM_GENERATION + '2'
+
+VIRT_HYPERV_SECURE_BOOT = _VIRT_HYPERV_NS + 'secure_boot'
+VIRT_HYPERV_SHIELDED_VMS =_VIRT_HYPERV_NS + 'shielded_vms'
+
+# RemoteFX suffix
+_VIRT_HYPERV_REMOTEFX = _VIRT_HYPERV_NS + 'remotefx:'
+
+# VRAM is requested in MB
+_VIRT_HYPERV_REMOTEFX_MEMORY = _VIRT_HYPERV_REMOTEFX + 'vram:'
+VIRT_HYPERV_REMOTEFX_MEMORY_64 = _VIRT_HYPERV_REMOTEFX_MEMORY + '64'
+VIRT_HYPERV_REMOTEFX_MEMORY_128 = _VIRT_HYPERV_REMOTEFX_MEMORY + '128'
+VIRT_HYPERV_REMOTEFX_MEMORY_256 = _VIRT_HYPERV_REMOTEFX_MEMORY + '256'
+VIRT_HYPERV_REMOTEFX_MEMORY_512 = _VIRT_HYPERV_REMOTEFX_MEMORY + '512'
+VIRT_HYPERV_REMOTEFX_MEMORY_1024 = _VIRT_HYPERV_REMOTEFX_MEMORY + '1024'
```

Comment:
This is a quantitative resource class, not a qualitative trait. As such, you should add a standard resource class to the nova/objects/fields.py:ResourceClass STANDARD enumeration for VIDEO_MEMORY_MB.